Enterprise-CMCS/macpro-mdct-mfp
# Fix for Add, Update & Delete of Objective & Funding Source

### Description

There is an issue currently in the data structure for add/edit/delete of cards in the initiative dashboard section for Objective & Funding Source due to the nesting of the forms.

OLD: Only 1 objective was able to be saved because the data was saving at the initiative level
![old](https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/127151429/047a697c-c782-49af-9ee0-774f8a39b8f2)


NEW: There is now an array stored inside initiatives using entityType as the object key
![new](https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/127151429/5feac504-2897-4238-bc4f-6285b72f810d)

NOTE: ignore the getEntityStatus.tsx changes, they will be use for the entity status later. This code snuck in but won't cause any problems if merged in now

Will be addressed in another PR:
- There is an issue with saved nested data not being render in the form when you try to edit the form after saving. If you reload the initiative dashboard, it will show up. You can see it in Objective when adding quantitive targets.
- Save & Return button not doing anything
- Code coverage will be increased later in separate ticket

### Related ticket(s)


CMDCT-193

---
### How to test

1) Sign into MFP
2) Create a new WP
3) Inside the new WP, go to "State and Territory-Specific Initiatives"
4) Add Initiative
5) Click "Edit" on the newly added Initiative 
6) In Evaluation Plan or Funding Sources, test adding, editing and deleting card. There should not be any issues when doing those actions and multiple of each should be able to coexist

### Important updates


N/A

---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
